yosys 0.60 tests fail with `ERROR: posix_spawn abc failed` when `ABCEXTERNAL:=abc` is set

### Version

0.60

### On which OS did this happen?

Linux

### Reproduction Steps

Hi @KrystalDelusion I'm following up from https://github.com/YosysHQ/yosys/issues/5537 with a reproducer for the behaviour of `make test` failing like this when trying to build yosys 0.60 with `ABCEXTERNAL:=abc` under Alpine Linux 3.23.0

We need an Alpine build environment, which requires generating a couple of temporary keys first:

```
docker run --name keys --entrypoint abuild-keygen --env PACKAGER="Yosys <yosys@test.org>" strophy/alpine-abuild:3.23.0 -n
```

You will see the key name with some random, e.g.: `/home/builder/.abuild/yosys@test.org-693d24c3.rsa`. Copy the keys to your local `~/.abuild` dir and prepare the aports repo with my yosys branch:

```
mkdir ~/.abuild
docker cp keys:/home/builder/.abuild/yosys@test.org-693d24c3.rsa ~/.abuild/
docker cp keys:/home/builder/.abuild/yosys@test.org-693d24c3.rsa.pub ~/.abuild/
git clone -b yosys-0.60 https://gitlab.alpinelinux.org/strophy/aports.git
cd aports
```

Start the build environment (mount the keys you generated earlier by name):

```
docker run \
-v "$PWD:/home/builder/package" \
-v "$HOME/.abuild/packages:/packages" \
-v "$HOME/.abuild/yosys@test.org-693d24c3.rsa.pub:/etc/apk/keys/yosys@test.org-693d24c3.rsa.pub" \
-v "$HOME/.abuild/yosys@test.org-693d24c3.rsa.rsa.pub:/home/builder/.abuild/yosys@test.org-693d24c3.rsa.pub" \ 
-v "$HOME/.abuild/yosys@test.org-693d24c3.rsa:/home/builder/.abuild/yosys@test.org-693d24c3.rsa" \ 
-e PACKAGER_PRIVKEY="/home/builder/.abuild/badfunkstripe@gmail.com-69131905.rsa" \ 
-it --entrypoint bash \
strophy/alpine-abuild:3.23.0
```

Then run the full build:

```
cd testing/yosys
abuild -r
```

The build should succeed, but tests should fail. Observe errors like:
```
Test: arrays02 -> ERROR!
Note: Make sure that 'iverilog' is an up-to-date git checkout of Icarus Verilog.
make[1]: *** [../tools/autotest.mk:10: arrays02.sv] Error 1
```

View test error logs:
```
cat src/tests/simple/arrays02.err
> ERROR: posix_spawn abc failed
which abc
> /usr/bin/abc
```

Modify the APKBUILD file on your host by e.g. commenting out `ABCEXTERNAL:=abc`, run the build again, and the tests should pass. You can also re-run individual steps like this:
```
abuild clean
abuild deps
abuild unpack
abuild prepare
abuild build
abuild check
```


Clean up by deleting the container and `~/.abuild` directory. Thanks for your time and help investigating this, I hope the process is not too involved or time consuming!

### Expected Behavior

yosys should build and test against external abc without errors

### Actual Behavior

Confusing errors about iverilog appear to be masking an underlying issue finding the system `abc` binary, even though this in the path and can be seen after running `abuild deps` with `which abc`

--- a/passes/techmap/abc.cc
+++ b/passes/techmap/abc.cc
@@ -252,8 +252,8 @@ std::optional<AbcProcess> spawn_abc(const char* abc_exe, DeferredLogs &logs) {
 
 	char arg1[] = "-s";
 	char* argv[] = { strdup(abc_exe), arg1, nullptr };
-	if (0 != posix_spawn(&result.pid, abc_exe, &file_actions, nullptr, argv, environ)) {
-		logs.log_error("posix_spawn %s failed", abc_exe);
+	if (0 != posix_spawnp(&result.pid, abc_exe, &file_actions, nullptr, argv, environ)) {
+		logs.log_error("posix_spawnp %s failed", abc_exe);
 		return std::nullopt;
 	}
 	free(argv[0]);

